tests/conftest.py

```diff
@@ -905,6 +906,12 @@ def __call__(
                 else:
                     DEFAULT_DATE = timezone.datetime(2016, 1, 1)
                     self.start_date = DEFAULT_DATE
+
+            # Handle deprecated argument `concurrency` for DAG class
+            # If "concurrency" is defined, replace it with "max_active_tasks"
+            if concurrency:
+                self.kwargs["max_active_tasks"] = concurrency
+                self.kwargs.pop("concurrency")
```

Comment:
I would rather replace it in place rather than hide it.

We could raise an error in case if provide deprecated parameters into this factory (this could be done by the separate PR)

Reply:
Thanks @Taragolis . I think directly replacing it in place will result in failures from other tests so I've raised a warning instead. I believe completely deprecating the `concurrency` param requires a larger PR to look at all the tests

Reply:
I think it could be resolved pretty easy, at least same things we do when try to resolve

- You add a raise en error
- I've add `full test needed`
- We retrieve list of all tests where it failed
- After 1 or 2 iteration it could be finally resolved

This only about a tests, so we do not need to deprecate it into the tests, because tests do not have any versions and could be changed in any time, and better change it as soon as it possible for avoid situation, that we forget about this changes

Reply:
I just analyse [latest warnings summary](https://github.com/apache/airflow/files/15209519/test-summarized-warnings.9.zip) and found that there are only two tests at all where concurrency uses in `dag_maker`

```console
tests/jobs/test_scheduler_job.py::TestSchedulerJob::test_infinite_pool
tests/jobs/test_scheduler_job.py::TestSchedulerJob::test_not_enough_pool_slots
```

Reply:
So feel free to raise an error e.g RuntimeError (with description, that this argument is deprecated in AIrflow) and replace in particular this test.
We still run in full tests, but just in case